cap_anndata/cap_anndata.py

```diff
@@ -120,6 +122,53 @@ def _validate_cap_df(self, cap_df: CapAnnDataDF, axis: int) -> None:
                 "AnnData object!"
             )
 
+    def _link_array_mapping(self, cap_dict: CapAnnDataDict, key: str) -> None:
+        if key not in self._file.keys():
+            raise KeyError(f"The key {key} doesn't exist in the file! Ignore linking.")
+
+        group = self._file[key]
+        if not isinstance(group, h5py.Group):
+            raise ValueError(f"The object {key} must be a group!")
+
+        for array_name in group.keys():
+            array = group[array_name]
+            if isinstance(array, h5py.Dataset):
+                cap_dict[array_name] = array
+            elif isinstance(array, h5py.Group):
+                cap_dict[array_name] = sparse_dataset(array)
+            else:
+                raise ValueError(f"Can't link array in {key} due to unsupported type of object: {type(array)}")
+
+    def _create_new_matrix(
+            self,
+            dest: str,
+            matrix: Union[np.ndarray, ss.csr_matrix, ss.csc_matrix, None] = None,
+            matrix_shape: Union[tuple[int, int], None] = None,
+            data_dtype: Union[np.dtype, None] = None,
+            format: Union[str, None] = None,
+            compression: str = "lzf"
+    ) -> None:
+        if matrix is not None:
+            self._write_elem(dest, matrix, compression=compression)
+        else:
+            if format == "dense":
+                group = self._file.create_dataset(name=dest, shape=matrix_shape, dtype=data_dtype,
+                                                  compression=compression)
+                # https://anndata.readthedocs.io/en/latest/fileformat-prose.html#dense-arrays-specification-v0-2-0
+                group.attrs['encoding-type'] = 'array'
+                group.attrs['encoding-version'] = '0.2.0'
+            elif format in ["csr",
```

Comment:
Need add todo to incapsulate them to some Enum (with `dense` to)

Reply:
done